Sources/Classes/Views/CustomizableButton/CustomizableButtonView.swift

```diff
@@ -0,0 +1,355 @@
+//
+//  Copyright (c) 2019 Touch Instinct
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the Software), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED AS IS, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import RxSwift
+import RxCocoa
+
+public typealias Spinner = UIView & Animatable
+
+public struct CustomizableButtonState: OptionSet {
+
+    // MARK: - OptionSet conformance
+
+    public let rawValue: Int
+
+    public init(rawValue: Int) {
+        self.rawValue = rawValue
+    }
+
+    // MARK: - States
+
+    public static let highlighted = CustomizableButtonState(rawValue: 1 << 1)
+    public static let normal = CustomizableButtonState(rawValue: 1 << 2)
+    public static let enabled = CustomizableButtonState(rawValue: 1 << 3)
+    public static let disabled = CustomizableButtonState(rawValue: 1 << 4)
+    public static let loading = CustomizableButtonState(rawValue: 1 << 5)
+
+    // MARK: - Properties
+
+    public var isLoading: Bool {
+        return contains(.loading)
+    }
+}
+
+open class CustomizableButtonView: UIView {
+
+    // MARK: - Stored Properties
+
+    private let disposeBag = DisposeBag()
+    private let button = CustomizableButton()
```

Comment:
maybe it's better to keep it public, like UIButton.titleLabel do. think about it.

Reply:
I don't see any benefits making button public. You want access to UIButton.titleLabel to constraint some of your views relative to that label, for example, but in CustomizableButtonVIew's appearance you can configure almost what you want.